crates/polars-plan/src/logical_plan/optimizer/cse_expr.rs

```diff
@@ -684,7 +684,7 @@ impl<'a> RewritingVisitor for CommonSubExprOptimizer<'a> {
                     schema,
                     options,
                 } => {
-                    let input_schema = alp.schema(lp_arena);
+                    let input_schema = lp_arena.get(*input).schema(lp_arena);
```

Comment:
I wonder if `ALogicalPlan::Aggregate` also need the same fix? 🤔

Reply:
You are right. Missed that one.

Reply:
I have an issue unrelated to this change, but I would like to receive quick confirmation here:

Do you think that `col ("b").sum() + col("a").sum().over ([col ("b")]) + col ("b").sum()` can be reused for `col("b").sum()`. If so, I would open an issue.